edk2toollib/uefi/authenticated_variables_structure_support.py

```diff
@@ -79,21 +97,32 @@ def PopulateFromFileStream(self, fs, decodesize):
 
         self.SignatureOwner = uuid.UUID(bytes_le=fs.read(16))
 
-        # read remainling decode size for x509 data
+        # read remaining decode size for x509 data
         self.SignatureDataSize = decodesize - EfiSignatureDataEfiCertX509.STATIC_STRUCT_SIZE
-        self.SignatureData = memoryview(fs.read(self.SignatureDataSize))
-
-    def Print(self):
-        print("EfiSignatureData - EfiSignatureDataEfiCertX509")
-        print("  Signature Owner:      %s" % str(self.SignatureOwner))
-        print("  Signature Data: ")
-        if(self.SignatureData is None):
-            print("    NONE")
+        self.SignatureData = (fs.read(self.SignatureDataSize))
+
+    def Print(self, compact: bool = False):
```

Comment:
Strictly speaking, this concept of "compact" vs "not compact" aligns almost identically with how `__str__()` and `__repr__()` are supposed to be defined. May we should switch to that?

Reply:
https://medium.com/swlh/string-representations-in-python-understand-repr-vs-str-12f046986eb5

Reply:
Eh, maybe not so much, but food for thought.

Reply:
That was educational.  Here, the compact form is not intended to assist code debugging, but rather a human readable format that lends itself to automated post processing.  As such, `__repr__()` does not make sense.  `__str__()` might be nice, we could implement it by calling Print(), but... maybe next time, since my focus is on the compact form, and str does not aid that.